packages/mantine/src/components/table/Table.tsx

```diff
@@ -231,11 +232,17 @@ export const Table: TableType = <T,>({
     }, []);
 
     useDidUpdate(() => {
-        triggerChange();
-        if (!multiRowSelectionEnabled) {
-            clearSelection();
-        }
-    }, [state.globalFilter, state.sorting, state.pagination, form.values]);
+        const debounceTable = debounce(() => {
+            triggerChange();
+            if (!multiRowSelectionEnabled) {
+                clearSelection();
+            }
+        }, 500);
```

Comment:
I'm wondering if that causes issue because the debounced method is overridden each time anything changes in the state (maybe my solution is not better)

I was thinking of something like:

```typescript
const triggerChange = debounce(() => onChange?.({...state, ...form.values}), 500);

useDidUpdate(() => {
    triggerChange();
    if (!multiRowSelectionEnabled) {
        clearSelection();
    }
}, [state.globalFilter, state.sorting, state.pagination, form.values]);

useEffect(() => {
    return () => triggerChange.cancel();
}, []);
```

Reply:
I've put the cancel in the useEffect on line `230` for less code duplication (which has the same dependency array as you suggested)